### PR TITLE
Add Cygwin CI and stop installing unwanted plugins

### LINKED ISSUE #2396: NetCDF 4.9.0: undefined reference to `ncp_win32_api' on MinGW (libncpoco)
I tried to update the NetCDF version available to julia users.
However, netCDF 4.9.0 fails to build during cross-compilation for  x86_64-w64-mingw32 with the error "undefined reference to `ncp_win32_api'" . 

* the version of the software with which you are encountering an issue

NetCDF 4.9.0

* environmental information (i.e. Operating System, compiler info, java version, python version, etc.)

target x86_64-w64-mingw32 (cross compiled from Linux)
With gcc version: x86_64-w64-mingw32-gcc (GCC) 4.8.5

* a description of the issue with the steps needed to reproduce it

Here are the steps how we compile NetCDF (https://github.com/JuliaPackaging/Yggdrasil/blob/master/N/NetCDF/common.jl#L59)

```bash
./configure --prefix=/workspace/destdir --build=x86_64-linux-musl --host=x86_64-w64-mingw32 --enable-shared --disable-static --disable-dap-remote-tests --disable-testsets
make LDFLAGS="-L/workspace/destdir/bin -lhdf5-0 -lhdf5_hl-0 -lcurl-4 -lz -no-undefined -Wl,--export-all-symbols" -j8
```

The build step fails with:

```
[22:59:58] make[2]: Entering directory '/workspace/srcdir/netcdf-c-4.9.0/liblib'           
[22:59:58] /bin/sh ../libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I..  -I../include -I../include -I../oc2 -I../libnczarr  -I../libhdf5  -I../libdap2 -I
../oc -I../libdap4 -I../libncxml -I../libsrc4 -I../libnczarr -I../libncpoco -I/workspace/destdir/include  -std=c99 -fno-strict-aliasing -MT libnetcdf_la-nc_initialize
.lo -MD -MP -MF .deps/libnetcdf_la-nc_initialize.Tpo -c -o libnetcdf_la-nc_initialize.lo `test -f 'nc_initialize.c' || echo './'`nc_initialize.c
[22:59:58] libtool: compile:  cc -DHAVE_CONFIG_H -I. -I.. -I../include -I../include -I../oc2 -I../libnczarr -I../libhdf5 -I../libdap2 -I../oc -I../libdap4 -I../libncx
ml -I../libsrc4 -I../libnczarr -I../libncpoco -I/workspace/destdir/include -std=c99 -fno-strict-aliasing -MT libnetcdf_la-nc_initialize.lo -MD -MP -MF .deps/libnetcdf
_la-nc_initialize.Tpo -c nc_initialize.c  -DDLL_EXPORT -DPIC -o .libs/libnetcdf_la-nc_initialize.o
[22:59:58] mv -f .deps/libnetcdf_la-nc_initialize.Tpo .deps/libnetcdf_la-nc_initialize.Plo                                                     
[22:59:58] /bin/sh ../libtool  --tag=CC   --mode=link cc  -std=c99 -fno-strict-aliasing -version-info 20:0:1  -L/workspace/destdir/bin -lhdf5-0 -lhdf5_hl-0 -lcurl-4 -
lz -no-undefined -Wl,--export-all-symbols -o libnetcdf.la -rpath /workspace/destdir/lib libnetcdf_la-nc_initialize.lo ../libdispatch/libnetcdf2.la ../libdispatch/libd
ispatch.la ../libsrc/libnetcdf3.la  ../libhdf5/libnchdf5.la  ../libdap2/libdap2.la ../oc2/liboc.la ../libdap4/libdap4.la ../libncxml/libncxml.la ../libsrc4/libnetcdf4
.la ../libnczarr/libnczarr.la  ../libncpoco/libncpoco.la -lm -lsz -lxml2 -lhdf5-0 -lhdf5_hl-0 -lcurl-4 -lz                             
[22:59:58] libtool: link: cc -shared  .libs/libnetcdf_la-nc_initialize.o  -Wl,--whole-archive ../libdispatch/.libs/libnetcdf2.a ../libdispatch/.libs/libdispatch.a ../
libsrc/.libs/libnetcdf3.a ../libhdf5/.libs/libnchdf5.a ../libdap2/.libs/libdap2.a ../oc2/.libs/liboc.a ../libdap4/.libs/libdap4.a ../libncxml/.libs/libncxml.a ../libs
rc4/.libs/libnetcdf4.a ../libnczarr/.libs/libnczarr.a ../libncpoco/.libs/libncpoco.a -Wl,--no-whole-archive  -L/workspace/destdir/bin -lsz -lxml2 -lhdf5-0 -lhdf5_hl-0
 -lcurl-4 -lz  -Wl,--export-all-symbols   -o .libs/libnetcdf-19.dll -Wl,--enable-auto-image-base -Xlinker --out-implib -Xlinker .libs/libnetcdf.dll.a                 
[22:59:58] ../libncpoco/.libs/libncpoco.a(ncpoco.o):ncpoco.c:(.text+0x40): undefined reference to `ncp_win32_api'
[22:59:58] ../libncpoco/.libs/libncpoco.a(ncpoco.o):ncpoco.c:(.text+0x4e): undefined reference to `ncp_win32_api'                                                     
[22:59:58] ../libncpoco/.libs/libncpoco.a(ncpoco.o):ncpoco.c:(.text+0x5c): undefined reference to `ncp_win32_api'
[22:59:58] ../libncpoco/.libs/libncpoco.a(ncpoco.o):ncpoco.c:(.text+0x6a): undefined reference to `ncp_win32_api'                                                     
[22:59:58] ../libncpoco/.libs/libncpoco.a(ncpoco.o):ncpoco.c:(.text+0x78): undefined reference to `ncp_win32_api'
[22:59:58] ../libncpoco/.libs/libncpoco.a(ncpoco.o):ncpoco.c:(.text+0x86): more undefined references to `ncp_win32_api' follow                                        
[22:59:58] /opt/x86_64-w64-mingw32/bin/../lib/gcc/x86_64-w64-mingw32/4.8.5/../../../../x86_64-w64-mingw32/bin/ld: ../libncpoco/.libs/libncpoco.a(ncpoco.o): bad reloc 
address 0x0 in section `.pdata'                                                                                                                                       
[22:59:58] collect2: error: ld returned 1 exit status
[22:59:58] make[2]: *** [Makefile:612: libnetcdf.la] Error 1                                                                                                          
[22:59:58] make[2]: Leaving directory '/workspace/srcdir/netcdf-c-4.9.0/liblib'
[22:59:58] make[1]: *** [Makefile:764: all-recursive] Error 1                                                                                                         
[22:59:58] make[1]: Leaving directory '/workspace/srcdir/netcdf-c-4.9.0'                                                                                              
[22:59:58] make: *** [Makefile:609: all] Error 2                                                                   
[22:59:58]  ---> make LDFLAGS="${LDFLAGS_MAKE}" -j${nproc}                                                                                                            
[22:59:58]  ---> make LDFLAGS="${LDFLAGS_MAKE}" -j${nproc}                                                                       
[22:59:58] Previous command exited with 2                                                                                                                             
[22:59:58] Child Process exited, exit code 2              
┌ Warning: Build failed, the following log files were generated:                                  
│   - ${WORKSPACE}/srcdir/netcdf-c-4.9.0/config.log                                        
```

Attached is the file [config.log](https://github.com/Unidata/netcdf-c/files/8888640/config.log)

Thank you for your help.

### LINKED ISSUE #2468: Link error when compiling NCZhdf5filters.o on MinGW64 with netCDF 4.9.0
I am still trying to update NetCDF for Julia users on Windows. In this test, I tried native compilation on Windows (using the MinGW64 toolchain from MSYS2).

NetCDF is configured as:

```
./configure --disable-testsets  --enable-shared  --disable-static  --disable-dap-remote-tests
make LDFLAGS=" -no-undefined -Wl,--export-all-symbols"
```

The MinGW64 compiler version is:

```
$ gcc --version
gcc.exe (Rev2, Built by MSYS2 project) 12.1.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

The full error message is:

```
make[3]: Entering directory '/home/Alexander Barth/netcdf-c-2/plugins'
/bin/sh ../libtool  --tag=CC   --mode=link gcc  -fno-strict-aliasing   -module -avoid-version -shared -export-dynamic -rpath /home/Alexander Barth/netcdf-c-2/plugins -no-undefined -no-undefined -Wl,--export-all-symbols -o lib__nczhdf5filters.la -rpath /home/Alexander Barth/netcdf-c-2/plugins/plugindir NCZhdf5filters.lo  -lhdf5_hl -lhdf5 -lm -lz -lsz -lbz2 -lzstd -lxml2 -lcurl
libtool: link: rm -fr  .libs/lib__nczhdf5filters.dll.a
libtool: link: gcc -shared  .libs/NCZhdf5filters.o   -lhdf5_hl -lhdf5 -lz -lsz -lbz2 -lzstd -lxml2 -lcurl  -Wl,--export-all-symbols   -o .libs/lib__nczhdf5filters.dll -Wl,--enable-auto-image-base -Xlinker --out-implib -Xlinker .libs/lib__nczhdf5filters.dll.a
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: .libs/NCZhdf5filters.o:NCZhdf5filters:(.text+0x1a24): undefined reference to `nc_inq_var'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: .libs/NCZhdf5filters.o:NCZhdf5filters:(.text+0x1a5a): undefined reference to `nc_inq_type'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: .libs/NCZhdf5filters.o:NCZhdf5filters:(.text+0x2350): undefined reference to `nc_inq_var'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: .libs/NCZhdf5filters.o:NCZhdf5filters:(.text+0x2386): undefined reference to `nc_inq_type'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: .libs/NCZhdf5filters.o:NCZhdf5filters:(.text+0x2423): undefined reference to `nc_inq_dimlen'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: .libs/NCZhdf5filters.o:NCZhdf5filters:(.text+0x245a): undefined reference to `nc_inq_var_chunking'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: .libs/NCZhdf5filters.o:NCZhdf5filters:(.text+0x24ee): undefined reference to `nc_inq_var_endian'
collect2.exe: error: ld returned 1 exit status
make[3]: *** [Makefile:872: lib__nczhdf5filters.la] Error 1
make[3]: Leaving directory '/home/Alexander Barth/netcdf-c-2/plugins'
make[2]: *** [Makefile:731: all] Error 2
make[2]: Leaving directory '/home/Alexander Barth/netcdf-c-2/plugins'
make[1]: *** [Makefile:764: all-recursive] Error 1
make[1]: Leaving directory '/home/Alexander Barth/netcdf-c-2'
make: *** [Makefile:609: all] Error 2
```

These steps worked when compiling NetCDF 4.8.1 on the same machine. 

It might be related to the LDFLAGS passed to make, which is  also used here:
https://github.com/Unidata/netcdf-c/blob/main/.github/workflows/run_tests_win_mingw.yml#L49

Any help would be greatly appreciated!

Maybe this is related:
https://github.com/Unidata/netcdf-c/issues/2419

### PR DESCRIPTION
**Pull request by DWesl**
This adds a CI job for my development platform, which will hopefully avoid issues like #2196 and #2446.

This also avoids installing test plugins (or any plugins if `--without-plugin-dir` is passed to configure), which is especially noticed by distribution package managers using a `DESTDIR` install: that issue was previously raised as #2431. 

I tried to get CMake CI runs working on Cygwin and MinGW, but couldn't get any tests to pass on Cygwin or complete the build on MinGW; this is likely due to the problems highlighted in #2290.

Includes cross-compilation suggestion [from this comment](https://github.com/Unidata/netcdf-c/issues/2396#issuecomment-1172799403).

Also includes a few markdown cleanups that seem to have been fixed in other ways since I started working; let me know if you want to approach those changes differently

Replaces #2431; replaces #2440
Might close #2468
Might replace #2479 
Might close #2396

### CODE REVIEW

**Comment by DennisHeimbigner** [comment]
This PR will take some time to digest,
but thanks for doing this.

File: .github/workflows/run_tests_win_mingw.yml
Comment (DennisHeimbigner) [inline .github/workflows/run_tests_win_mingw.yml:10]: The "push" here should be removed.

File: .github/workflows/run_tests_win_mingw.yml
Comment (DennisHeimbigner) [inline .github/workflows/run_tests_win_mingw.yml:49]: Why remove the -no-undefined?

File: .gitignore
Comment (DennisHeimbigner) [inline .gitignore:31]: Are you sure that test--driver and ltmain.sh should be ignored?

File: configure.ac
Comment (DennisHeimbigner) [inline configure.ac:105]: I vaguely recall that we had problems in the past using AC_CANONICAL_HOST.
Wish I could remember the details.

File: libncpoco/cp_win32.c
Comment (DennisHeimbigner) [inline libncpoco/cp_win32.c:32]: Not sure what is the point of this change. Also, I believe we have tried to
avoid use of inline because older versions of some C compilers do not support it.

File: plugins/Makefile.am
Comment (DennisHeimbigner) [inline plugins/Makefile.am:140]: Have you tested this on the platforms we support?

File: .github/workflows/run_tests_win_mingw.yml
Comment (DWesl) [reply .github/workflows/run_tests_win_mingw.yml:49]: It's already in the `Makefile.am`, and I like having the flags necessary for things to build actually specified by the build system.

I tried removing `-Wl,--export-all-symbols` as well, but that led to a number of "unresolved reference to symbol nc..." errors at link time.  I might be able to make more progress by specifying `AM_CPPFLAGS=-DDLL_EXPORT -DDLL_NETCDF` in the top-level `Makefile.am` so the source files properly specify `__declspec(dllexport)`, but given that CMake fails to link I'm not sure that would be enough.

File: .gitignore
Comment (DWesl) [reply .gitignore:31]: For the developers' git repository? Almost certainly.  For release tarballs (made with `make dist` or `make distcheck`), no, they should absolutely be included.

On a related note, what is the policy on building GitHub's automatic repository tarballs?  Do you assume everyone has a full Autotools or CMake install (the assumption I made here) or only POSIX Make and Bourne Shell (the assumption made to Autotools release tarballs)?

File: configure.ac
Comment (DWesl) [reply configure.ac:105]: It won't auto-discover MSVC, but it should help with #2396, since the current assumption made in using `uname` directly is that nobody will want to cross-compile at any time for any reason

File: libncpoco/cp_win32.c
Comment (DWesl) [reply libncpoco/cp_win32.c:32]: The point of this change is defining a fallback definition of `nulldup` in case `strdup` is not provided by the C runtime.  I don't remember if that was a problem on Cygwin or MinGW, but the current definitions all assume that `strdup` will be provided if `nulldup` isn't.

Comment (DWesl) [reply libncpoco/cp_win32.c:32]: I tried adding `inline`, but since it wasn't in the declaration (or the other twenty definitions) the C compiler rejected it and I took it out of the definition.

File: plugins/Makefile.am
Comment (DWesl) [reply plugins/Makefile.am:140]: I've tested this on Cygwin and MinGW [on my fork](https://github.com/DWesl/netcdf-c/tree/cygwin-plugin-build) and locally.

I think you've also tested this, since this is basically the solution proposed in #2440. 

I have a test for what this is trying to do (fix #2431) in the Cygwin CI (`make install DESTDIR=/tmp/pretend-root && test ! -d /tmp/pretend-root/$(pwd)`, more or less); I could try to add a similar test to the `installcheck` target, but that might take a while.

File: .github/workflows/run_tests_win_mingw.yml
Comment (DWesl) [reply .github/workflows/run_tests_win_mingw.yml:10]: Done, here and in the Cygwin workflow.

File: libncpoco/cp_win32.c
Comment (DennisHeimbigner) [reply libncpoco/cp_win32.c:32]: Good point about strdup; you might consider putting the body of the
alternate nulldup into a more general location such as libdispatch/dutil.c
or putting it into ncconfigure.h as a static function.

**Comment by DennisHeimbigner** [comment]
OK, I think this is ok to merge subject to testing with Visual Studio.

**Comment by DWesl** [comment]
Do you want me to try to add MSVC testing to CI as part of this PR?  CMake should be straightforward; Autotools will need help (specifically, someone telling me what values to specify for `CC`, `CPP`, and `LD`)

**Comment by DennisHeimbigner** [comment]
I do note that the Mac build is currently failing for filters.

**Comment by DWesl** [comment]
It's passing in #2527 just before the merge, and on #2519, which is the most likely place to have broken things.  Given [this comment](https://github.com/Unidata/netcdf-c/issues/2478#issuecomment-1263852033), the only thing I can think of is that I added `AM_LDFLAGS` to each plugins' `_LDFLAGS` and maybe [`-avoid-version`](https://github.com/DWesl/netcdf-c/blob/4046afdfc2ee28fc5369ea4af08365c078cdfc77/plugins/Makefile.am#L7) doesn't play nicely with [`-version-info 0:0:0`](https://github.com/DWesl/netcdf-c/blob/4046afdfc2ee28fc5369ea4af08365c078cdfc77/plugins/Makefile.am#L34)

**Comment by DWesl** [comment]
Apparently `-version-info` is required for plugins on MacOS and breaks plugins on MinGW and Cygwin, which is not what I expected

File: libncpoco/cp_win32.c
Comment (DWesl) [reply libncpoco/cp_win32.c]: Moved to `include/ncconfigure.h`

**Comment by WardF** [comment]
Reviewing extant PR's to pull into v4.9.1-rc2; thanks a lot, I just noticed the checks hadn't run. Running now.

File: .gitignore
Comment (WardF) [reply .gitignore:31]: We turn on maintainer mode in the release branches, so full installs should not be required.  A lot of the generated files are explicitly kept out of version tracking, and only added to the 'v[xyz]-release' branch which gets tagged.

**Comment by WardF** [comment]
@DennisHeimbigner checks are passing, but you are more plugged into the cygwin ecosystem.  Any blocking issues I'm not seeing, here?  Otherwise I'll merge it later this afternoon :).

**Review by WardF — APPROVED** [review]

**Comment by DennisHeimbigner** [comment]
I think my previous review holds, so it should be ok to merge it.

File: .gitignore
Comment (DWesl) [reply .gitignore:31]: This will help keep the generated files out of version tracking, but will make adding them to the release branches a bit trickier unless the release branches drop this part of the `.gitignore` file; should I add a comment at the start to that effect?

E.g.
```gitignore
# Autotools-generated files
# Delete this section on release branches
```

**Comment by WardF** [comment]
Thanks @DennisHeimbigner !

**Comment by DennisHeimbigner** [comment]
I am revisiting this because it is causing some problems.
Specifically, I do not understand why github action run_tests_cygin.yml contains the code below.
What is its purpose?
> if [ -d "/tmp/pretend-root/$(pwd)" ];
          then
              find /tmp/pretend-root/$(pwd)
              if [ $(find /tmp/pretend-root/$(pwd) -type f | wc -l) -gt 0 ]; then exit 1; fi
          fi

**Comment by DWesl** [comment]
> I am revisiting this because it is causing some problems.

Could you elaborate on what problems you run into?  I'm not finding any cases where only the Cygwin workflow fails, only ones that also fail on Ubuntu.

**Comment by DennisHeimbigner** [comment]
The problem is that noinst causes my test filters to not get built at test time, hence those tests fail.
I am modifying the Makefile.am to use test_LTLIBRARIES and testdir instead. But am running up against
the above piece of code during github action execution.

**Comment by DWesl** [comment]
Found it here:
https://github.com/Unidata/netcdf-c/compare/main...DennisHeimbigner:netcdf-c:s3lib.tmp#diff-72bb278809ac7dbe0329f064d73cbe73a5dda1d45ea85c288f5b9d3f3cbc07fbL39-R43
you tell Automake you do not have any test libraries, no libraries you don't want to install, only libraries you want to install to the source directory.  Here
https://github.com/Unidata/netcdf-c/compare/main...DennisHeimbigner:netcdf-c:s3lib.tmp#diff-72bb278809ac7dbe0329f064d73cbe73a5dda1d45ea85c288f5b9d3f3cbc07fbL123-R125
you tell Automake that you do, in fact, want to install the test plugins, to a rather odd installation directory under what would be the source directory if it were installed at root on the same computer.

The test is checking whether Autotools is installing anything to that rather odd installation directory, and fails, as it is supposed to, when it finds that Autotools has installed the test plugins to that directory, rather than not installing them at all, as you likely intended.  

The test is there because many distribution package managers use `DESTDIR` to install to an empty directory so that creating an archive to install on another computer looks like `cd ${DESTDIR}; tar caf ../netcdf-c-${VERSION}.tar.gz *`.  They would rather not install the test plugins, not least because the people who install the package from their archive would not have the rest of the files to run the tests.

When you change from a target that installs to one that doesn't (i.e. `check_LTLIBRARIES`, which runs at `make check` time, or `noinst_LTLIBRARIES`, which runs at `make` time), you will need to add `-rpath /tmp` or similar to the `_LDFLAGS` of the test plugins, otherwise Autotools will assume the library is for linking into a test executable and not bother creating a shared library ([after all](https://github.com/Unidata/netcdf-c/pull/2431#issuecomment-1169259449), the flag for saying "I want a shared library I can `dlopen` later" [is `-rpath` rather than `-shared`](https://www.gnu.org/software/automake/manual/html_node/Libtool-Convenience-Libraries.html), because that isn't the least bit confusung to anyone)

**Comment by DWesl** [comment]
> The problem is that noinst causes my test filters to not get built at test time, hence those tests fail. 

That is very odd, since `noinst_LTLIBRARIES` builds at the same time as everything else (i.e. when you run `make`).  I can't see any reason that wouldn't run before `make check`, though you could try `check_LTLIBRARIES`, which would build the library before running the tests.

Can I see the test logs where `noinst_LTLIBRARIES` fails?

**Comment by DennisHeimbigner** [comment]
Nevertheless. It does not work.

**Comment by DennisHeimbigner** [comment]
If there was some way to force DESTDIR to not be applied to the testdir, then things would work ok, I think.

**Comment by DWesl** [comment]
> Nevertheless. It does not work.

The only thing I can think of is maybe someone added new test plugins and forgot to add `-rpath /tmp` to their `_LDFLAGS`, so Autotools just build a static library, which the test couldn't `dlopen`.

It would be more convenient if Autotools had a sensible flag for to tell it we're building a test plugin, rather than it deciding `-rpath` is the flag to use to tell it "I want to build something I can `dlopen` later, not just something I can link to a test executable".

**Comment by DennisHeimbigner** [comment]
AFAIK using your solution, the test filters never get built at make check time in any form, static or dynamic.

**Comment by DWesl** [comment]
> AFAIK using your solution, the test filters never get built at make check time in any form, static or dynamic.

That is very odd.  I'd like to check this; where are the plugin tests that are failing?  I put all test plugins in `check_LTLIBRARIES`, so any of the plugin tests should work, then I can just view the logs for similar failures from the workflow runs on any PR.

The other option is reverting this PR and using the original version of #2431, allowing Autotools to install the files to wherever [and then removing them right after](https://github.com/Unidata/netcdf-c/pull/2431#issuecomment-1167770994). `/tmp/testplugins` might be a convenient install directory for the test plugins, since the tests use the versions in the current directory.

I still like the "don't install the test plugins in the first place" option, which is why I made this PR.  It did take a fair bit of digging (discussion around this from [here](https://github.com/Unidata/netcdf-c/pull/2431#issuecomment-1168157090) to [here](https://github.com/Unidata/netcdf-c/pull/2431#issuecomment-1169259449) to find out that just adding the test libraries to `check_LTLIBRARIES` wouldn't build the shared libraries the tests need, since the tests `dlopen` the plugins rather than linking to them; you need to add `-rpath` to the corresponding `lib__plugin_name_la_LDFLAGS` variables, which is done automatically for all `_LTLIBRARIES` variables except `check_LTLIBRARIES` and `noinst_LTLIBRARIES`.

**Comment by DWesl** [comment]
I'm attempting to reproduce this with your branch `s3lib.tmp`, and the following configure line:
```bash
../configure --enable-doxygen --enable-dot --disable-static --enable-shared -C --no-create --no-recursion --enable-s3 --enable-s3-internal --enable-nczarr-s3-tests --enable-nczarr --enable-nczarr-filters --enable-byterange --enable-hdf5 --enable-dap --enable-dap-remote-tests --enable-logging
```

The first problem I run into is that something in `libdispatch` uses SSL functions directly, and appears to assume that linking against `libcurl` will bring those functions in at runtime, and all will be well.  However, this breaks the assumptions of the `-no-undefined` libtool flag, which states that all functions will be resolvable at link time and is required on Windows (and therefore Cygwin).

I have jury-rigged shims in there that should let me compile, but I am unsure how to do this properly or generally; perhaps [Autoconf Archive's `ax_check_openssl`](https://www.gnu.org/software/autoconf-archive/ax_check_openssl.html) would help, at least for platforms that are neither Windows or Apple?

**Comment by DennisHeimbigner** [comment]
I fixed this in PR https://github.com/Unidata/netcdf-c/pull/2658.

**Comment by DennisHeimbigner** [comment]
Good idea.

**Comment by DWesl** [comment]
Now #2659.

Looks like I was wrong.  Still no idea what was wrong or why that fixed things.

**Pull request merged**
